app/javascript/packs/application.js

```diff
@@ -0,0 +1,17 @@
+// This file is automatically compiled by Webpack, along with any other files
+// present in this directory. You're encouraged to place your actual application logic in
+// a relevant structure within app/javascript and only use these pack files to reference
+// that code so it'll be compiled.
+
+require("@rails/ujs").start()
```

Comment:
file này dùng làm gfi

Reply:
UJS là một kỹ thuật không đưa mã JS của bạn vào đánh dấu HTML, nhưng trong ngữ cảnh Rails cụ thể, đó là một thư viện JavaScript nhỏ do nhóm Rails phát triển.Chúng có tác dụng :
hộp thoại xác nhận trong liên kết hoặc biểu mẫu
phương thức  not-GET trong các hành động liên kết
các nút tự động vô hiệu hóa trong các biểu mẫu để ngăn chặn double submitting
các biểu mẫu và liên kết từ xa - các cuộc gọi và phản hồi AJAX tự động tiến hành
bảo vệ CSRF tự động trong các yêu cầu AJAX